### DISCUSSION TITLE
Does TimingEngine.time_at() incorporate the simfile offset?

### DISCUSSION THREAD

**Discussion by ifeelagood** [garcia/simfile#35] [General]
Hi, thank you for the module! For some code that I am writing, I need to align note time with song time. the simfile object has the attribute "offset", which as I understand it is the length of time until beat 0. Are the note times when using time_notes() relative to the "song" or the actual audio?

thank you

**Comment by garcia** [comment]
The timing engine initializes beat 0's time set to the negative of the timing data offset:

https://github.com/garcia/simfile/blob/main/simfile/timing/engine.py#L211-L224

I think this is the same as what you would see in the StepMania 5 editor. For files that are properly synced, `time_notes()` should return the correct time for the audio file.

**Reply by ifeelagood** [reply]
Apologies for creating an issue instead of a discussion! 

That's amazingly reassuring; I'm using simfile for a machine learning project, so one less point of potential failure.

p.s. thanks so much for simfile! It has saved me probably weeks...
